### PR TITLE
switch to monkeypatch.chdir

### PR DESCRIPTION
**Pull request by AlexVCaron**
# Quick description

Cryptic error raised by #954. In tests, we switch to a temporary directory, which gets deleted when a test module is done and its context is tear down. However, `os.chdir` does not change the directory back after the test case. In some corner cases, the next test case does not change its directory, and thus still lives in the temporary directory of another one. In even deeper corner cases, the temporary directory gets deleted before the test case runs. In that case, any call to `os.getcwd()` fails, such as the one done by `pytest-console-script`, leading to [this failure](https://github.com/scilus/scilpy/actions/runs/8457802675/job/23170612617#step:7:228).

I switched to `monkeypatch.chdir` a pytest fixture which does the switching back after the test ends.

...

## Type of change

Check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by EmmaRenauld — APPROVED** [review]
Let's approve and see if my PR works after rebasing!

**Comment by AlexVCaron** [comment]
I'm confident in the changes. I'll let you merge, since it's my PR

**Pull request merged**
